app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/model/DefaultSampleInfoRepository.kt

```diff
@@ -47,51 +48,51 @@ object DefaultSampleInfoRepository : SampleInfoRepository {
             // List that will be populated with samples
             val sampleList = mutableListOf<Sample>()
-            // Iterate through the metadata folder for all metadata files
-            context.assets.list("samples")?.forEach { samplePath ->
-                // Get this metadata files as a string
-                context.assets.open("samples/$samplePath/README.metadata.json").use { inputStream ->
-                    val metadataJsonString = inputStream.bufferedReader().use { it.readText() }
-                    try {
-                        val metadata = json.decodeFromString<SampleMetadata>(metadataJsonString)
 
-                        // Create and add a new sample metadata data class object to the list
-                        val sample = Sample(
-                            name = metadata.title,
-                            codeFiles = Sample.loadCodeFiles(
-                                context = context,
-                                sampleName = metadata.title
-                            ),
-                            url = "https://developers.arcgis.com/kotlin/sample-code/" +
-                                    metadata.title.replace(" ", "-").lowercase(),
-                            readMe = loadReadMe(
-                                context = context,
-                                sampleName = samplePath
-                            ),
-                            screenshotURL = loadScreenshot(
-                                sampleName = metadata.title,
-                                imageArray = metadata.imagePaths
-                            ),
-                            mainActivity = loadActivityPath(
-                                codePaths = metadata.codePaths
-                            ),
-                            metadata = metadata,
-                        )
-                        // Add the new sample to the list
-                        sampleList.add(sample)
-                    } catch (e: Exception) {
-                        Log.e(
-                            DefaultSampleInfoRepository::class.simpleName,
-                            "Exception at $samplePath: " + e.printStackTrace()
-                        )
-                    }
+            // Read the entire 'samples.json' from build/sampleAssets/samples/
+            val samplesJsonString = context.assets.open("samples/samples.json").use { stream ->
+                stream.bufferedReader().use { it.readText() }
+            }
+
+            // Parse it into a map of: sampleFolderName -> mapOf( filename -> fileContent )
+            val allSamplesData = json.decodeFromString<Map<String, Map<String, String>>>(samplesJsonString)
+
+            // Build each Sample using the metadata from "README.metadata.json"
+            allSamplesData.forEach { (sampleFolderName, fileMap) ->
+                val metadataJsonString = fileMap["README.metadata.json"] ?: return@forEach
```

Comment:
If we're getting a null for the metadata, that'd be a big enough problem we probably want to throw